### PR TITLE
Fix/bad blocks v5

### PR DESCRIPTION
**Pull request by MarekM25**
## Changes
Fixing all issues related to block production. Our block production fuzzer is finally green. I will test it on Sepolia Validators. Engine hive tests are passing. 

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by MarekM25** [comment]
> Still want the trace change from the original PR?
> 
> ```diff
> - if (!options.ContainsFlag(ProcessingOptions.ForceProcessing))
> + if (!options.ContainsFlag(ProcessingOptions.Trace))
> ```

Yes :)

**Comment by benaadams** [comment]
Lot of AuRa tests failing with issue with 'dueTime'?
```
System.ArgumentOutOfRangeException :
Number must be either non-negative and less than or equal to Int32.MaxValue or -1. (Parameter 'dueTime')
```

**Comment by MarekM25** [comment]
> Lot of AuRa tests failing with issue with 'dueTime'?
> 
> ```
> System.ArgumentOutOfRangeException :
> Number must be either non-negative and less than or equal to Int32.MaxValue or -1. (Parameter 'dueTime')
> ```

Okay, I know what's going on, will fix it soon :)

File: src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V1.PayloadProduction.cs [hunk 1]
```diff
@@ -4,10 +4,14 @@
...
 using System.Collections;
 using System.Collections.Generic;
+using System.Data;
+using System.IO;
 using System.Linq;
+using System.Security.Cryptography;
 using System.Threading;
 using System.Threading.Tasks;
 using FluentAssertions;
+using Google.Protobuf.WellKnownTypes;
```
Comment (benaadams) [inline src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V1.PayloadProduction.cs:14]: Lots of weird new `using`s?

Comment (MarekM25) [reply src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V1.PayloadProduction.cs]: Removed, thx

**Pull request merged**